File: src/models/ChatMessage.ts
import { Class } from "./Class";
import WritableDbItem from "./WritableDbItem";

export default interface ChatMessage extends WritableDbItem {
  /**
   * The emergency associated with the chat message
   * */
  emergencyId: string;

  /**
   * The user id of the message sender
   * */
  senderId: string;

  /**
   * The rsiHandle of the message sender
   * */
  senderRsiHandle: string;

  /**
   * The contents of the message
   * */
  contents: string;

  /**
   * Whether the message has been edited
   * */
  edited: boolean;

  /**
   * Whether the message has been deleted
   * */
  deleted: boolean;

  /**
   * The Medrunner Class of the message sender at the time of sending
   * */
  senderClass: Class;
}

File: src-dotnet/models/ChatMessage.cs
namespace MedrunnerApiClient.Models;

/// <summary>
/// Represents a chat message.
/// </summary>
public class ChatMessage : DbItem
{
    /// <summary>
    /// The emergency associated with the chat message.
    /// </summary>
    public string EmergencyId { get; set; }

    /// <summary>
    /// The user id of the message sender.
    /// </summary>
    public string SenderId { get; set; }

    /// <summary>
    /// The timestamp at which the message was sent in Unix seconds.
    /// </summary>
    public long MessageSentTimestamp { get; set; }

    /// <summary>
    /// The contents of the message.
    /// </summary>
    public string Content { get; set; }
}
